hazelcast/src/main/java/com/hazelcast/client/impl/statistics/ClientStatisticsService.java

```diff
@@ -128,28 +132,30 @@ private TcpClientConnection getConnection() {
      * @param periodSeconds the interval at which the statistics collection and send is being run
      */
     private void schedulePeriodicStatisticsSendTask(long periodSeconds) {
+        client.getTaskScheduler().scheduleWithRepetition(this::collectAndSendStats, 0, periodSeconds, SECONDS);
+    }
+
+    private void collectAndSendStats() {
         ClientMetricCollector clientMetricCollector = new ClientMetricCollector();
         CompositeMetricsCollector compositeMetricsCollector = new CompositeMetricsCollector(clientMetricCollector,
                 publisherMetricsCollector);
 
-        client.getTaskScheduler().scheduleWithRepetition(() -> {
-            long collectionTimestamp = System.currentTimeMillis();
-            metricsRegistry.collect(compositeMetricsCollector);
-            publisherMetricsCollector.publishCollectedMetrics();
+        long collectionTimestamp = System.currentTimeMillis();
+        metricsRegistry.collect(compositeMetricsCollector);
+        publisherMetricsCollector.publishCollectedMetrics();
 
-            TcpClientConnection connection = getConnection();
-            if (connection == null) {
-                logger.finest("Cannot send client statistics to the server. No connection found.");
-                return;
-            }
+        TcpClientConnection connection = getConnection();
+        if (connection == null) {
+            logger.finest("Cannot send client statistics to the server. No connection found.");
+            return;
+        }
 
-            final StringBuilder clientAttributes = new StringBuilder();
-            periodicStats.fillMetrics(collectionTimestamp, clientAttributes, connection);
-            addNearCacheStats(clientAttributes);
+        final StringBuilder clientAttributes = new StringBuilder();
```

Comment:
Just asking... what's the deal with `final` here?

Reply:
I think it means constant variable but I don't actually know why we don't use final everywhere that's possible to use. Maybe there's something special to StringBuilder